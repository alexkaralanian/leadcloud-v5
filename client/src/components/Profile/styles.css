.profilePic {
  display: flex;
  margin: 20px 0px;
}

.logoutBtn {
  height: 45px;
  width: 175px;
  border-radius: 2px;
}

.logoutBtn span {
  font-weight: 200;
  letter-spacing: 0.125rem;
  margin-right: -0.125rem;
}
