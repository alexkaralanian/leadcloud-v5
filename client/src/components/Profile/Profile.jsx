import React from "react";
import PropTypes from "prop-types";
import { Grid, Col, Row, Image } from "react-bootstrap";
import { profilePic } from "./styles.css";
import { headerText } from "../../index.css";
// import { LogoutButton, GetGmailProfile } from "../Buttons/AuthButtons";

const Profile = ({ logout, user }) => (
  <Grid>
    <Row>
      <Col xs={12}>
        <div>
          <h1 className={headerText}>Welcome, {user.firstName}!</h1>
          <Image src={user.googlePhoto} className={profilePic} />
          <div>
            <button onClick={logout}>LOGOUT</button>
            {/* <GetGmailProfile onClick={getGmailProfile} /> */}
          </div>
        </div>
      </Col>
    </Row>
  </Grid>
);

Profile.propTypes = {
  logout: PropTypes.func.isRequired,
  user: PropTypes.object.isRequired
};

Profile.defaultProps = {
  user: {},
  logout: () => {}
};

export default Profile;
