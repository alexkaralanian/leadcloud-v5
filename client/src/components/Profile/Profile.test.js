import React from "react";
import { shallow, configure } from "enzyme";
import Profile from "./Profile";
import Adapter from "enzyme-adapter-react-16";
configure({ adapter: new Adapter() });

test("Profile Component renders correctly", () => {
  const component = shallow(<Profile />);
  expect(component).toMatchSnapshot();
});
