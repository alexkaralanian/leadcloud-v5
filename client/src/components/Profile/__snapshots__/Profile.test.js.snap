// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Profile Component renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Profile
    logout={[Function]}
    user={Object {}}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "bsClass": "container",
      "children": <Row
        bsClass="row"
        componentClass="div"
      >
        <Col
          bsClass="col"
          componentClass="div"
          xs={12}
        >
          <div>
            <h1
              className="headerText"
            >
              Welcome, 
              !
            </h1>
            <Image
              bsClass="img"
              circle={false}
              className="profilePic"
              responsive={false}
              rounded={false}
              src={undefined}
              thumbnail={false}
            />
            <div>
              <button
                onClick={[Function]}
              >
                LOGOUT
              </button>
            </div>
          </div>
        </Col>
      </Row>,
      "componentClass": "div",
      "fluid": false,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "bsClass": "row",
        "children": <Col
          bsClass="col"
          componentClass="div"
          xs={12}
        >
          <div>
            <h1
              className="headerText"
            >
              Welcome, 
              !
            </h1>
            <Image
              bsClass="img"
              circle={false}
              className="profilePic"
              responsive={false}
              rounded={false}
              src={undefined}
              thumbnail={false}
            />
            <div>
              <button
                onClick={[Function]}
              >
                LOGOUT
              </button>
            </div>
          </div>
        </Col>,
        "componentClass": "div",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "bsClass": "col",
          "children": <div>
            <h1
              className="headerText"
            >
              Welcome, 
              !
            </h1>
            <Image
              bsClass="img"
              circle={false}
              className="profilePic"
              responsive={false}
              rounded={false}
              src={undefined}
              thumbnail={false}
            />
            <div>
              <button
                onClick={[Function]}
              >
                LOGOUT
              </button>
            </div>
          </div>,
          "componentClass": "div",
          "xs": 12,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h1
                className="headerText"
              >
                Welcome, 
                !
              </h1>,
              <Image
                bsClass="img"
                circle={false}
                className="profilePic"
                responsive={false}
                rounded={false}
                src={undefined}
                thumbnail={false}
              />,
              <div>
                <button
                  onClick={[Function]}
                >
                  LOGOUT
                </button>
              </div>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "Welcome, ",
                  undefined,
                  "!",
                ],
                "className": "headerText",
              },
              "ref": null,
              "rendered": Array [
                "Welcome, ",
                undefined,
                "!",
              ],
              "type": "h1",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "bsClass": "img",
                "circle": false,
                "className": "profilePic",
                "responsive": false,
                "rounded": false,
                "src": undefined,
                "thumbnail": false,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <button
                  onClick={[Function]}
                >
                  LOGOUT
                </button>,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "LOGOUT",
                  "onClick": [Function],
                },
                "ref": null,
                "rendered": "LOGOUT",
                "type": "button",
              },
              "type": "div",
            },
          ],
          "type": "div",
        },
        "type": [Function],
      },
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "bsClass": "container",
        "children": <Row
          bsClass="row"
          componentClass="div"
        >
          <Col
            bsClass="col"
            componentClass="div"
            xs={12}
          >
            <div>
              <h1
                className="headerText"
              >
                Welcome, 
                !
              </h1>
              <Image
                bsClass="img"
                circle={false}
                className="profilePic"
                responsive={false}
                rounded={false}
                src={undefined}
                thumbnail={false}
              />
              <div>
                <button
                  onClick={[Function]}
                >
                  LOGOUT
                </button>
              </div>
            </div>
          </Col>
        </Row>,
        "componentClass": "div",
        "fluid": false,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "bsClass": "row",
          "children": <Col
            bsClass="col"
            componentClass="div"
            xs={12}
          >
            <div>
              <h1
                className="headerText"
              >
                Welcome, 
                !
              </h1>
              <Image
                bsClass="img"
                circle={false}
                className="profilePic"
                responsive={false}
                rounded={false}
                src={undefined}
                thumbnail={false}
              />
              <div>
                <button
                  onClick={[Function]}
                >
                  LOGOUT
                </button>
              </div>
            </div>
          </Col>,
          "componentClass": "div",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "bsClass": "col",
            "children": <div>
              <h1
                className="headerText"
              >
                Welcome, 
                !
              </h1>
              <Image
                bsClass="img"
                circle={false}
                className="profilePic"
                responsive={false}
                rounded={false}
                src={undefined}
                thumbnail={false}
              />
              <div>
                <button
                  onClick={[Function]}
                >
                  LOGOUT
                </button>
              </div>
            </div>,
            "componentClass": "div",
            "xs": 12,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <h1
                  className="headerText"
                >
                  Welcome, 
                  !
                </h1>,
                <Image
                  bsClass="img"
                  circle={false}
                  className="profilePic"
                  responsive={false}
                  rounded={false}
                  src={undefined}
                  thumbnail={false}
                />,
                <div>
                  <button
                    onClick={[Function]}
                  >
                    LOGOUT
                  </button>
                </div>,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    "Welcome, ",
                    undefined,
                    "!",
                  ],
                  "className": "headerText",
                },
                "ref": null,
                "rendered": Array [
                  "Welcome, ",
                  undefined,
                  "!",
                ],
                "type": "h1",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "bsClass": "img",
                  "circle": false,
                  "className": "profilePic",
                  "responsive": false,
                  "rounded": false,
                  "src": undefined,
                  "thumbnail": false,
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <button
                    onClick={[Function]}
                  >
                    LOGOUT
                  </button>,
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "LOGOUT",
                    "onClick": [Function],
                  },
                  "ref": null,
                  "rendered": "LOGOUT",
                  "type": "button",
                },
                "type": "div",
              },
            ],
            "type": "div",
          },
          "type": [Function],
        },
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
